{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "authorship_tag": "ABX9TyNsDgtgH52XvSjnYb/Uf7Rp",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/carlocamilloni/Structural-Bioinformatics/blob/main/Notebooks/t09_design.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# PROTEIN DESIGN"
      ],
      "metadata": {
        "id": "mVRXfEzV_m6p"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "In this task you will use a protein design pipeline based on the combination of **RFDiffusion**, a generative model that can diffuse protein like backbone configurations. **ProteinMPNN** that given a backbone conformation can design a sequence that will fold into it. **AlphaFold2** to assess if the resulting protein structure is predicted consistently with the desired design. In this instruction file you will focus on producing an homoligomer, while for the report you will try to produce a de-novo binder as well as scaffolding an existing protein motif into a new protein."
      ],
      "metadata": {
        "id": "kh-kBDhKB3tk"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "The pipeline is implemented here: https://colab.research.google.com/github/sokrypton/ColabDesign/blob/main/rf/examples/diffusion.ipynb#scrollTo=pZQnHLuDCsZm"
      ],
      "metadata": {
        "id": "I9HHmE1RCMt-"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Exercise: Design a new protein forming large homo-oligomers"
      ],
      "metadata": {
        "id": "WrGX8PmCCLKN"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "The key options for this exercise are:\n",
        "\n",
        "```\n",
        "contigs=#AA  : this is the number of aminoacid in the monomer\n",
        "order=N : make N copies of the defined contig(s) and add a symmetry constraint, for homo-oligomeric diffusion.\n",
        "symmetry='cyclic' : this defines 360Â°/N symmetry\n",
        "```\n",
        "\n",
        "if at the end of the procedure the **RFdesign** and **AF2** prediction do not match it indicates that the design has likely failed. What you can do is either try to generate more **ProteinMPNN** sequences given the same design changing parameters like:\n",
        "\n",
        "```\n",
        "\n",
        "```\n",
        "\n",
        "\n",
        "\n"
      ],
      "metadata": {
        "id": "O9ToXITiCYGQ"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Exercise 2: Design of a binder"
      ],
      "metadata": {
        "id": "_oueIGzTEmUe"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "In the third exercise we will try to design protein binders for b2-microglobulin, an immunoglobulin protein that is part of our immune system.\n",
        "```\n",
        "contigs: A:100\n",
        "pdb: 5cs7      \n",
        "```\n",
        "\n"
      ],
      "metadata": {
        "id": "uX3_4sC5MEsz"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "The above line means that we want to bind chain A from the pdb 5cs7 (b2-microglobulin), and build a binder of 100 aa. Select also the \"beta-model\" and increase the number of designs to 4. In the proteinMPNN section you need to select:\n",
        "```\n",
        "initial_guess=True\n",
        "num_recycles=3\n",
        "use_multimer=True\n",
        "```\n",
        "\n",
        "If you do not get anything good, try to either generate more designs or to modify the length of the different segments. Do not make them too long to avoid the program to become too slow."
      ],
      "metadata": {
        "id": "t3DiF5obM7H6"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Exercise 3: Design a protein that can scaffold an active site"
      ],
      "metadata": {
        "id": "txcSlJxSMAho"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "Here we want to try to design the active site of the HIV-1 protease, that is encoded in a homodimeric protein (pdb: 1hiv), into a single protein chain.\n",
        "\n",
        "In the notebook above we need to provide some indications, for example in the \"contigues\" and \"pdb\" fields:\n",
        "```\n",
        "contigs: 35/A25-27/15/B25-27/35\n",
        "pdb: 1hiv\n",
        "```\n",
        "The above line means start your protein with 35 AA, then add the region 25-27 from chain A of the 1HIV model, then 15 more AA, then the region 25-27 from chain B of the 1HIV model. Finally add 35 more AA. The selected region from 1HIV is the one of the catalytic aspartic acids.\n",
        "\n",
        "Select also the \"beta-model\" and increase the number of designs to 4.\n",
        "\n",
        "If you do not get anything good, try to either generate more designs or to modify the length of the different segments. Do not make them too long to avoid the program to become too slow."
      ],
      "metadata": {
        "id": "Ltg08I7vR05q"
      }
    }
  ]
}