{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "authorship_tag": "ABX9TyM7JM17Xg1mT8GnfvSRQgRp",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/carlocamilloni/Structural-Bioinformatics/blob/main/Notebooks/t09_design.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# PROTEIN DESIGN"
      ],
      "metadata": {
        "id": "mVRXfEzV_m6p"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "In this task you will use a protein design pipeline based on the combination of **RFDiffusion**, a generative model that can diffuse protein like backbone configurations. **ProteinMPNN** that given a backbone conformation can design a sequence that will fold into it. **AlphaFold2** to assess if the resulting protein structure is predicted consistently with the desired design. In this instruction file you will focus on producing an homoligomer, while for the report you will try to produce a de-novo binder as well as scaffolding an existing protein motif into a new protein."
      ],
      "metadata": {
        "id": "kh-kBDhKB3tk"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "The pipeline is implemented here: https://colab.research.google.com/github/sokrypton/ColabDesign/blob/main/rf/examples/diffusion.ipynb#scrollTo=pZQnHLuDCsZm"
      ],
      "metadata": {
        "id": "I9HHmE1RCMt-"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Exercise: Design a new protein forming large homo-oligomers"
      ],
      "metadata": {
        "id": "WrGX8PmCCLKN"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "The key options for this exercise are:\n",
        "\n",
        "```\n",
        "contigs=#AA  : this is the number of aminoacid in the monomer\n",
        "order=N : make N copies of the defined contig(s) and add a symmetry constraint, for homo-oligomeric diffusion.\n",
        "symmetry='cyclic' : this defines 360Â°/N symmetry\n",
        "```\n",
        "\n",
        "if at the end of the procedure the **RFdesign** and **AF2** prediction do not match it indicates that the design has likely failed. What you can do is either try to generate more **ProteinMPNN** sequences given the same design changing parameters like:\n",
        "\n",
        "```\n",
        "mpnn_sampling_temp=T : this goes between 0 and 1, where higher increases the diversity in the sequence space explored\n",
        "use_solubleMPNN=Y/N : this enables the use of different model weights\n",
        "```\n",
        "\n",
        "if these are not enough you can restart the design and increase\n",
        "```\n",
        "iterations=N : number of iteration per design\n",
        "num_designs=n : number of generated designs\n",
        "```\n",
        "do this keeping in mind the increased required time.\n",
        "\n"
      ],
      "metadata": {
        "id": "O9ToXITiCYGQ"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Additional options for the design of a binder"
      ],
      "metadata": {
        "id": "_oueIGzTEmUe"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "When designing a binder you need to define your target protein, this is done here by providing a PDB identifier (for example 5cs7 for beta-2-microglobulin) and in contigs indicating first which chain(s) to use:\n",
        "```\n",
        "contigs: A:70\n",
        "pdb: 5cs7      \n",
        "```\n",
        "The above line means that we want to bind chain A from the pdb 5cs7, and build a binder of 70 aa.\n"
      ],
      "metadata": {
        "id": "uX3_4sC5MEsz"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "When doing a binder you should do some changes also to the **AF2** step:\n",
        "```\n",
        "initial_guess=True\n",
        "num_recycles=3\n",
        "use_multimer=True\n",
        "```\n",
        "\n",
        "These options allow AF2 to use the PDB structure in input, increase the prediction accuracy and focus on a multimeric interaction."
      ],
      "metadata": {
        "id": "t3DiF5obM7H6"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Additional options for the design a protein that can scaffold an active site"
      ],
      "metadata": {
        "id": "txcSlJxSMAho"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "When you want to design a protein scaffold around an existing protein site you need again to provide the protein and indicate which parts to reuse and what to do:\n",
        "```\n",
        "contigs: 35/A25-27/35\n",
        "pdb: 1hiv\n",
        "```\n",
        "The above line means start your protein with 35 AA, then add the region 25-27 from chain A of the 1HIV model, then add 35 more AA."
      ],
      "metadata": {
        "id": "Ltg08I7vR05q"
      }
    }
  ]
}